src/api/File.js

```diff
@@ -202,6 +202,65 @@ class File extends Item {
         }
     }
 
+    /**
+     * Get the thumbnail url for a given BoxItem.  The function will attempt
+     * to fetch a jpg thumbnail of the given dimensions.  If this fails, the function
+     * will attempt to fectch a 1024x1024 png of the first page of the file as a fallback.
+     *
+     * @param {BoxItem} item - item whose thumbnail should be fetched
+     * @param {string} dimensions - desired dimensions of thumbnail. Acceptable dimensions
+     * for a jpg are: "32x32", "94x94", "160x160", "320x320", "1024x1024", "2048x2048".
+     * @param {Function} successCallback - function to call with the thumbnail url. The thumbnail
+     * url will be null if one could not be fetched.
+     * @return {Promise<void>}
+     */
+    async getFileThumbnail(item: BoxItem, dimensions: string, successCallback: Function): Promise<void> {
+        if (this.isDestroyed()) {
+            return;
+        }
+
+        // TODO: implement cache for recently fetched thumbnails
+
+        const { id } = item;
+        const newUrl = `${this.getUrl(id)}?fields=representations`;
+
+        const access_token = this.xhr.token;
+        if (!access_token) {
+            return;
+        }
+
+        const xhrOptions: Object = {
+            url: newUrl,
+            headers: {
+                // API will return first representation it finds, so 1024x1024 png is fallback.
+                'X-Rep-Hints': `[jpg?dimensions=${dimensions},png?dimensions=1024x1024]`,
```

Comment:
Seems like we're mixing the Thumbnails rep call with the Single page image rep call, do we need the png? Specifically, I see this in the docs `Therefore, the first time you request a single-page representation, you will notice the status is set to none and the metadata->pages property is not returned, which indicates that the representation is not yet generated and that you must try again after some time.`